src/main/java/org/openas2/processor/storage/MDNFileModule.java

```diff
@@ -29,15 +29,18 @@ public void handle(String action, Message msg, Map options) throws OpenAS2Except
 			throw new OpenAS2Exception("Message has no MDN");
 		}
 
-		try
+		if (getParameter(PARAM_FILENAME, false) != null)
```

Comment:
You can write:

String paramFileName = getParameter(PARAM_FILENAME, false);
if (paramFileName != null)
{
....
}